modules/azure/src/main/clojure/xtdb/azure/file_watch.clj

```diff
@@ -0,0 +1,88 @@
+(ns xtdb.azure.file-watch
+  (:require [clojure.data.json :as json]
+            [clojure.string :as string]
+            [clojure.tools.logging :as log])
+  (:import [com.azure.core.credential TokenCredential]
+           [com.azure.messaging.servicebus ServiceBusClientBuilder ServiceBusReceivedMessageContext ServiceBusErrorContext ServiceBusProcessorClient]
+           [com.azure.messaging.servicebus.administration ServiceBusAdministrationClient ServiceBusAdministrationClientBuilder]
+           [com.azure.storage.blob.models ListBlobsOptions BlobItem]
+           [com.azure.storage.blob BlobContainerClient]
+           [java.util NavigableSet UUID]
+           java.util.function.Consumer))
+
+(defn file-list-init [{:keys [^BlobContainerClient blob-container-client prefix]}  ^NavigableSet file-name-cache]
+  (let [list-blob-opts (cond-> (ListBlobsOptions.)
+                         prefix (.setPrefix prefix))
+        filename-list (->> (.listBlobs blob-container-client list-blob-opts nil)
+                           (.iterator)
+                           (iterator-seq)
+                           (mapv (fn [^BlobItem blob-item]
+                                   (subs (.getName blob-item) (count prefix)))))]
+    (.addAll file-name-cache filename-list)))
+
+(defn mk-short-uuid []
+  (subs (str (UUID/randomUUID)) 0 8))
+
+(defn setup-topic-subscription [{:keys [^TokenCredential azure-credential servicebus-namespace servicebus-topic-name]}]
+  (let [servicebus-admin-client (-> (ServiceBusAdministrationClientBuilder.)
+                                    (.credential (format "%s.servicebus.windows.net" servicebus-namespace)
```

Comment:
is this a different credential to the 'default credential provider'?

Reply:
Yeah, needs to point to the service bus namespace

Reply:
more that this seems to require a `TokenCredential`?

Reply:
Oh sorry - was confused by the quesiton, I think a 'TokenCredential' is just the generic class for all of the auth classes, what is being based in here is the default credential created in the azure module?